src/Sylius/Bundle/CoreBundle/Locale/TranslationLocaleProvider.php

```diff
@@ -37,7 +39,7 @@ public function __construct(LocaleContextInterface $localeContext)
      */
     public function getCurrentLocale()
     {
-        return $this->localeContext->getCurrentLocale();
+        return $this->container->get('sylius.context.locale')->getCurrentLocale();
```

Comment:
After a discussion with @pjedrzejewski we agreed that the best solution for this issue would be to move `getAvailableLocales` method to another service `AvailableTranslationLocalesProvider`. It will make `TranslationLocaleProvider` not dependent on the database so the circular dependency issue would be solved. What do you think about that?